### PR TITLE
Ajna Multiply Validations

### PR DESCRIPTION
**Pull request by zerotucks**
## Ticket URL

Please provide a link to ticket:
https://app.shortcut.com/oazo-apps/story/10201/reuse-validations-from-ajna-borrow-in-ajna-multiply

## Description of Changes

Please list the changes introduced by this PR:

- [x] errors and warning validations added to common multiply payload preparation function
- [x] errors and warning validations added to ajna open multiply which doesn't use common factor presently
- [x] refactored ajna open multiply to use common multiply payload prep function that includes shared validations
- [x] [NEW] create a multiply specific dust limit validation 👉 https://github.com/OasisDEX/oasis-borrow/pull/2566/files

## How to Test

Please provide instructions on how to test the changes in this PR:

* `cd packages/dma-contracts`
* `yarn hardhat test test/e2e/strategies/ajna/open.test.ts`
* `yarn hardhat test test/e2e/strategies/ajna/close.test.ts`
* `yarn hardhat test test/e2e/strategies/ajna/adjust.test.ts`

## PR Definition of Done

Please ensure the following requirements have been met before marking the PR as ready for review:

- [x] All checks are passing
- [x] PR is linked to a corresponding ticket
- [x] PR title is clear and concise
- [ ] Code has been self-reviewed and any fixes or improvements noted
- [ ] Documentation has been updated if necessary

### CODE REVIEW

**Comment by paszkowskiDamian** [comment]
Actually empty errors and warnings list made me think in what case it should say it is not possible to close your position. I think we should check how the position looks like when its in middle of liquidation as I guess then it will not be possible to close such position.

File: packages/dma-library/src/strategies/ajna/multiply/close.ts [hunk 1]
```diff
@@ -73,13 +73,22 @@ export const closeMultiply: AjnaCloseStrategy = async (args, dependencies) => {
...
 
+  // Validation
+  const errors = [
+    // Add as required...
+  ]
+
+  const warnings = [
+    // Add as required..
+  ]
+
```
Comment (paszkowskiDamian) [inline packages/dma-library/src/strategies/ajna/multiply/close.ts:83]: I guess we didn't consider the liquidation case, but let's refine that and we will add additional checks in separate story.

File: packages/dma-library/src/strategies/ajna/multiply/common.ts [hunk 2]
```diff
@@ -182,8 +188,8 @@ export function prepareAjnaMultiplyDMAPayload(
   )
 
   const isDepositingEth = areAddressesEqual(args.position.pool.collateralToken, dependencies.WETH)
-  const txAmount = amountFromWei(args.collateralAmount, TYPICAL_PRECISION)
-
+  // const txAmount = amountFromWei(args.collateralAmount, TYPICAL_PRECISION)
```
Comment (zerotucks) [inline packages/dma-library/src/strategies/ajna/multiply/common.ts]:
```suggestion
```

**Comment by zerotucks** [comment]
> Actually empty errors and warnings list made me think in what case it should say it is not possible to close your position. I think we should check how the position looks like when its in middle of liquidation as I guess then it will not be possible to close such position.

Yes. That's a good point. I'll add that one in.

File: packages/dma-library/src/strategies/ajna/multiply/close.ts (hunk 1 above)
Comment (zerotucks) [reply packages/dma-library/src/strategies/ajna/multiply/close.ts:83]: OK separate story it is.

**Review by paszkowskiDamian — APPROVED** [review]

**Pull request merged**